### ISSUE TITLE
Bulk insert fails when records have different sets of changed columns

### ISSUE THREAD

**Issue by duc0** [jOOQ/jOOQ#4402]
`InsertSetStep.set(record)` behavior was changed in https://github.com/jOOQ/jOOQ/commit/4c5fe868a2d61379c250d1308574987fc4e9ac63 to handle changed columns. This batch insert statement now fails if `record1` and `record2` have different sets of changed columns:

``` java
db.insertInto(SOME_TABLE)
  .set(record1).newRecord()
  .set(record2).newRecord()
  .returning(SOME_FIELD).fetch()
```

For batch inserts, should the old behavior be preserved (`InsertSetStep.set(record)` should  set all the columns instead of just the changed columns) ?  The records inserted may be created or updated differently and it could be hard to guarantee that they have the same set of changed columns.

**Comment by lukaseder** [comment]
Thanks a lot for reporting. This seems to be a regression, indeed - not only related to batch execution, but to "bulk" execution in general.

We'll have to look into this carefully. If SQL standard multi-row inserts are supported, then the `DEFAULT` keyword could be applied. If it is emulated using `INSERT .. SELECT`, the behaviour might not be exactly the same.

**Comment by lukaseder** [comment]
We'll implement this in jOOQ 3.10 without `DEFAULT` semantics. All records that have less changed fields than the maximum changed record will get their relevant fields padded with nulls.

**Comment by lukaseder** [comment]
The regression risk of this change is too high for it to be merged to 3.9 or older releases.
